src/lib/db-listeners/db-groups-listener.ts

```diff
@@ -57,19 +58,21 @@ export class DBGroupsListener {
 
     // ensure that the current user is not in more than 1 group and groups are not oversubscribed
     Object.keys(groups).forEach((groupId) => {
-      const groupUsers = groups[groupId].users || {};
-      const userKeys = Object.keys(groupUsers);
-      if (userKeys.indexOf(user.id) !== -1) {
+      const rawUsers = groups[groupId].users || {};
+      // rawUsers can get interpreted as an array instead of a map if user IDs are small (e.g. in demo mode)
+      // So, make sure to filter out empty array spaces and convert number IDs to strings for standardization
+      const groupUsers = map(rawUsers, (groupUser, uid) => ({uid: `${uid}`, user: groupUser}))
```

Comment:
There's probably a cleaner solution here, I'll put some more thought in...

Reply:
Seems like a fine solution